src/Compilers/CSharp/Portable/FlowAnalysis/VariablesDeclaredWalker.cs

```diff
@@ -122,16 +122,22 @@ public override BoundNode VisitLocalFunctionStatement(BoundLocalFunctionStatemen
             return base.VisitLocalFunctionStatement(node);
         }
 
-        public override void VisitForEachIterationVariable(BoundForEachStatement node)
+        public override void VisitForEachIterationVariables(BoundForEachStatement node)
         {
             if (IsInside)
             {
-                if (node.IterationVariableOpt != null)
+                var deconstructionAssignment = node.DeconstructionOpt?.DeconstructionAssignment;
+
+                if (deconstructionAssignment == null)
                 {
-                    _variablesDeclared.Add(node.IterationVariableOpt);
+                    // regular, not deconstructing, foreach declares exactly one iteration variable
+                    _variablesDeclared.Add(node.IterationVariables.Single());
+                }
+                else
+                {
+                    // deconstruction foreach declares multiple variables
+                   deconstructionAssignment.Left.VisitAllElements((x, self) => self.Visit(x), this);
```

Comment:
>deconstructionAssignment.Left.VisitAllElements((x, self) => self.Visit(x), this); [](start = 19, length = 81)

It is probably possible to have a region that doesn't include entire foreach, but includes portion of the deconstruction. This code won't be called in this case, but it probably should be. Please add relevant region analysis tests.